CMakeLists.txt

```diff
@@ -72,12 +72,16 @@ string(TOUPPER ${ONEDPL_BACKEND} ONEDPL_BACKEND)
 message(STATUS "Using parallel policies with ${ONEDPL_BACKEND} backend")
 string(TOLOWER ${ONEDPL_BACKEND} ONEDPL_BACKEND)
 
+# CMAKE_CXX_COMPILER and CMAKE_CXX_COMPILER_VERSION cannot be used because
+# CMake 3.19 and older will detect IntelLLVM compiler as CLang with CLang-specific version, see https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
+if (CMAKE_CXX_COMPILER MATCHES ".*(dpcpp-cl|dpcpp|icx-cl|icpx)(.exe)?$")
+    set(DPCPP_COMPLER TRUE)
+    execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_VARIABLE DPCPP_COMPILER_VERSION_RAW)
```

Comment:
For these variable names `DPCPP_COMPILER*`.  Since DPCPP is deprecated in favor of `icpx -fsycl`, would it be better to name these differently?  Possibly `ICPX_FSYCL_COMPILER*`.

Reply:
I am hesitant about `ICPX_FSYCL_COMPILER`, because `fsycl` is just an option, it does not determine a compiler and there is another relevant driver: `icx-cl`.
What do you think about `INTEL_LLVM_COMPILER` compiler instead? It will be aligned with the compiler id provided by new cmake versions.

Meanwhile, there is also a `clang++` driver. It is available when the compiler is built from [sources](https://github.com/intel/llvm). But cmake detects it as clang (according to my experiments).

Reply:
Yes, `INTEL_LLVM_COMPILER*` is a better name.